packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -898,23 +945,71 @@ export class EthModule {
   }
 
   private async _sendRawTransactionAction(rawTx: Buffer): Promise<string> {
+    // We validate that the tx is not legacy nor eip-2930 here
+    // because otherwise the catch logic below gets too tricky
+    if (rawTx[0] <= 0x7f && rawTx[0] !== 1) {
+      throw new InvalidArgumentsError(`Invalid transaction`);
```

Comment:
What would trigger this error? A new EIP-2929 tx that it's not 2930 (e.g., 1559)?

Reply:
Or a bunch of random data.

Reply:
Added a comment.